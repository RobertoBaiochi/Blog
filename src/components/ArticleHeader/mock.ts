import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

const { posts } = data;
const {
  id,
  attributes: { title, excerpt, cover, author, categories, createdAt },
} = posts.data[1];

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
} as unknown as ArticleHeaderProps;
