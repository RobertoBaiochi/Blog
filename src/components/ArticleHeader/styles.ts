import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
