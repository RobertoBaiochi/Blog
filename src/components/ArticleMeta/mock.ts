import { ArticleMetaProps } from './';
import { data } from '../../api/data.json';

const { posts } = data;

const {
  attributes: {
    createdAt,
    author: {
      data: {
        id: idAuthor,
        attributes: { displayName: displayNameAuthor, slug: slugAuthor },
      },
    },
    categories,
  },
} = posts.data[1];

export default {
  createdAt,
  author: {
    idAuthor,
    displayNameAuthor,
    slugAuthor,
  },
  categories,
} as unknown as ArticleMetaProps;

// import { ArticleMetaProps } from '.';

// export default {
//   createdAt: '2023-05-09T04:49:38.694Z',
//   author: {
//     id: '1',
//     displayName: 'Roberto Baiochi',
//     slug: 'roberto-baiochi',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'ReactJs',
//       slug: 'react-js',
//     },
//     {
//       id: '2',
//       displayName: 'NextJs',
//       slug: 'next-js',
//     },
//   ],
// } as ArticleMetaProps;
