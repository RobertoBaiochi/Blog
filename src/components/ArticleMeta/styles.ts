import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
