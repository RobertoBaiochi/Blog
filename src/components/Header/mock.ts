import { HeaderProps } from '.';
import { data } from '../../api/data.json';

const { setting } = data;
const {
  data: {
    id,
    attributes: { blogName, blogDescription },
  },
} = setting;

export default {
  id,
  blogName,
  blogDescription,
  logo: data.setting.data.attributes.logo.data,
} as unknown as HeaderProps;
