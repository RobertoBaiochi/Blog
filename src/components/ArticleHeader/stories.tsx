import { Meta, StoryFn } from '@storybook/react';
import ArticleHeader, { ArticleHeaderProps } from '.';

import mock from './mock';

export default {
  title: 'ArticleHeader',
  component: ArticleHeader,
  args: mock,
  argTypes: {
    createdAt: {
      control: {
        type: 'date',
      },
    },
  },
} as Meta<ArticleHeaderProps>;

export const Template: StoryFn<ArticleHeaderProps> = (args) => {
  return (
    <div style={{ maxWidth: '96rem', marginInline: 'auto' }}>
      <ArticleHeader {...args} />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero animi, ad
        cumque illo laudantium, ab quod et quos omnis perspiciatis odit
        dignissimos facilis vitae? Rerum culpa cupiditate labore quas
        dignissimos.
      </p>
    </div>
  );
};
