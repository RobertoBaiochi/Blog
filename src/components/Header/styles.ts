import styled, { css } from 'styled-components';

import { Title as HeadingHeader } from '../Heading/styles';

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.sizes.maxWidth};
    margin-inline: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacing.huge};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};

    ${HeadingHeader} {
      margin: 0 0 calc(${theme.spacing.small} - 3rem);
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    height: 100%;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacing.large};
  `};
`;
