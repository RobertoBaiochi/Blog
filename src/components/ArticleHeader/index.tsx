import { Author } from '../shared-typed/author';
import { Category } from '../shared-typed/category';
import { StrapiImage } from '../shared-typed/cover';
import { Wrapper, Excerpt, Cover } from './styles';

import Heading from '../Heading';
import ArticleMeta from '../ArticleMeta';

export type ArticleHeaderProps = {
  id: string;
  title: string;
  excerpt: string;
  cover: StrapiImage;
  author: Author;
  categories: Category[];
  createdAt: string;
};

const ArticleHeader = ({
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
}: ArticleHeaderProps) => {
  return (
    <Wrapper>
      <Heading as="h3" size="big">
        {title}
      </Heading>

      <Excerpt>{excerpt}</Excerpt>

      <Cover src={cover.url} alt={cover.alternativeText} />

      <ArticleMeta
        author={author}
        categories={categories}
        createdAt={createdAt}
      />
    </Wrapper>
  );
};
export default ArticleHeader;
