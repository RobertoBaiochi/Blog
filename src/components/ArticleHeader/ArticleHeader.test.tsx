import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import ArticleHeader, { ArticleHeaderProps } from '.';
import '@testing-library/jest-dom';
import mock from './mock';
import { formatDate } from '@/utils/format-date';

const props: ArticleHeaderProps = mock;

describe('<ArticleHeader />', () => {
  it('should render heading, excerpt, cover img and meta', () => {
    renderTheme(<ArticleHeader {...props} />);

    const { alternativeText } = props.cover.data[0].attributes;

    const title = screen.getByRole('heading', {
      name: props.title,
    });

    const img = screen.getByRole('img', {
      name: alternativeText,
    });

    const excerpt = screen.getByText(props.excerpt);

    const date = screen.getByText(formatDate(props.createdAt));

    expect(title).toBeInTheDocument();
    expect(img).toBeInTheDocument();
    expect(excerpt).toBeInTheDocument();
    expect(date).toBeInTheDocument();
  });

  it('should render heading, excerpt, cover img and meta', () => {
    const { container } = renderTheme(<ArticleHeader {...props} />);

    expect(container).toMatchSnapshot();
  });
});
