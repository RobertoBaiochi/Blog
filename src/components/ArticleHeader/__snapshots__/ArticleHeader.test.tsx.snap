// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should render heading, excerpt, cover img and meta 1`] = `
<div>
  <header
    class="sc-acde2419-0 guJeco"
  >
    <h3
      class="sc-b5aaf39e-0 jNSrfn"
    >
      Usando o Canvas para criar gráficos em JavaScript
    </h3>
    <p
      class="sc-acde2419-1 jRTpsP"
    >
      O artigo apresenta como usar a API Canvas em HTML e JavaScript para criar gráficos personalizados com estilização e interatividade, resultando em visualizações de dados impressionantes e envolventes.
    </p>
    <img
      alt="praia vista de cima, água cor vende escuro e rochas"
      class="sc-acde2419-2 hMevjM"
      src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683581086/adam_seckel_xhaz6_Nn_VT_64_unsplash_ad1c1db2e6.jpg"
    />
    <div
      class="sc-7c55c9a4-0 bRvXpo"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/marcela-strug"
        >
          Marcela Strug
        </a>
        <span
          class="separator"
        >
           | 
        </span>
        <time
          datetime="2023-05-09T04:33:33.718Z"
        >
          9 de mai. de 2023
        </time>
        <span
          class="separator"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/java-script"
            >
              JavaScript
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`;
